internal/dag/policy.go

```diff
@@ -665,16 +665,20 @@ func loadBalancerRequestHashPolicies(lbp *contour_api_v1.LoadBalancerPolicy, val
 		hashSourceIPSet := false
 		// Map of unique header names.
 		headerHashPolicies := map[string]bool{}
+		// Map of unique query parameter names.
+		queryParameterHashPolicies := map[string]bool{}
 		for _, hashPolicy := range lbp.RequestHashPolicies {
 			rhp := RequestHashPolicy{
 				Terminal: hashPolicy.Terminal,
 			}
 
 			// Ensure hashing for exactly one request attribute is set.
-			if (!hashPolicy.HashSourceIP && hashPolicy.HeaderHashOptions == nil) ||
-				(hashPolicy.HashSourceIP && hashPolicy.HeaderHashOptions != nil) {
+			if (!hashPolicy.HashSourceIP && hashPolicy.HeaderHashOptions == nil && hashPolicy.QueryParameterHashOptions == nil) ||
```

Comment:
another way to skin this is to have a counter for number of options set and individually check if each option is set, incrementing if so

if counter is 0 or greater than 1, then write the warning

we will probably have to do something like that if we add any more options and to simplify what we already have

Reply:
Done